### ISSUE TITLE
Generated preprocessed_configs/config.yaml is actually in XML format

### ISSUE THREAD

**Issue by pmed** [ClickHouse/ClickHouse#30584]
**How to reproduce**

0. Create a config.yaml file:

```yaml
max_server_memory_usage_to_ram_ratio: 0.5
```

1. Run ClickHouse with this configuration:

`clickhouse-local -C config.yaml --verbose`

>  ClickHouse % out/programs/clickhouse-local -C config.yaml --verbose
Processing configuration file 'config.yaml'.
Saved preprocessed configuration to './preprocessed_configs/config.yaml'.
Initialized background executor for merges and mutations with num_threads=16, num_tasks=32
Initialized background executor for move operations with num_threads=8, num_tasks=8
Initialized background executor for fetches with num_threads=8, num_tasks=8
Initialized background executor for common operations (e.g. clearing old parts) with num_threads=8, num_tasks=8
std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in create_directory: Permission denied [/var/lib/clickhouse]
Cannot print extra info for Poco::Exception (version 21.11.1.1)

2. It generates a `preprocessed_configs/config.yaml` that is actually an XML:
```xml
<!-- This file was generated automatically.
     Do not edit it: it is likely to be discarded and generated again before it's read next time.
     Files used to generate this file:
       config.yaml      -->
<clickhouse><max_server_memory_usage_to_ram_ratio>0.5</max_server_memory_usage_to_ram_ratio></clickhouse>
```

3. An attempt to start ClickHouse with the generated config leads to the error:

> Processing configuration file 'preprocessed_configs/config.yaml'.
Code: 585. DB::Exception: Unable to parse YAML configuration file preprocessed_configs/config.yaml, yaml-cpp: error at line 2, column 20: illegal map value. (CANNOT_PARSE_YAML), Stack trace (when copying this message, always include the lines below):

<Empty trace>
 (version 21.11.1.1)

* Which ClickHouse server version to use: version 21.11.1.1 (recently built from master branch)

**Expected behavior**

The generated config filename should have `.xml` extension, i.e. `preprocessed_configs/config.yaml.xml`, in order the ClickHouse can determinate the configuration format by the file extension:

> ClickHouse % out/programs/clickhouse-local -C preprocessed_configs/config.yaml.xml   
Processing configuration file 'preprocessed_configs/config.yaml.xml'.
Saved preprocessed configuration to './preprocessed_configs/config.yaml.xml'.
std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in create_directory: Permission denied [/var/lib/clickhouse]
Cannot print extra info for Poco::Exception (version 21.11.1.1)` 

**Error message and/or stacktrace**

For some reason clickhouse-local doesn't start on macOS Big Sur 11.6 with a custom config due to an exception:

> std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in create_directory: Permission denied [/var/lib/clickhouse]
Cannot print extra info for Poco::Exception (version 21.11.1.1)

**Comment by azat** [comment]
Seems that this had been fixed by #35020

**Comment by pmed** [comment]
Yes, it seems so, thanks! So I'm closing this.
